docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,309 @@
+convention, not by diktat.  Example parameters:
+   * Max instances - limit of number of volumes allowed
+   * Min free instances - how many ready-to-use volumes should be maintained
+   * Max free instances - limit of number of ready-to-use volumes
+   * Size distribution - desired histogram of sizes of ready-to-use volumes
+
+In addition to these, there will be volume-type specific parameters, such as
+the GCE SSD vs Disk param described above.
+
+### Scrubbing
+
+Volumes need to be scrubbed before they can be reused, and some storage devices
+may even need to be scrubbed before they are decommissioned.  This is the
+responsibility of the `VolumeManager` as it drives the lifecycle.
+
+## Networking
+
+TODO: flip Ingress and LB ?
+
+Network ingress in Kubernetes will be represented by `LoadBalancer`s.  A user
+can create an `IngressClaim` and reference that claim from an `Ingress` object.
+A `LoadBalancerManager` is responsible for binding `IngressClaim`s to
+`LoadBalancers`s.  If the `IngressClaim` can not be satisified by any existing
+`LoadBalancer`, the `LoadBalancerManager` may provision a new one.
+
+```
+                       +--------------+
+        +--create----> | LoadBalancer |
+        |              +--------------+
+   +----+----+               ^
+   |  LBMgr  |---bind------> :
+   +----+----+               v
+        |              +---------------+
+        +---watch----> | IngressClaim  | <--+
+                       +---------------+    |
+                                 ^          |
+                                 |          +-------- user-created
+                                 |          |
+                           +----------+     |
+                           | Ingress  | <---+
+                           +----------+
+```
+
+### IngressClaim
+
+An `IngressClaim` does not specify which specific Ingress it wants.  Instead it
+specifies characteristics of the ingress it needs and those characteristics are
+satisfied as best is possible.  Characteristics might include the availability
+properties required or the specific IP address to use.
+
+Like `PersistentVolumeClaim`, users will be able to request coarse "classes" of
+ingress (e.g. Gold, Silver, Bronze).  How those classes map to real storage
+technologies is a policy set by the cluster administrators.
+
+### LoadBalancerManagers
+
+The act of binding a `IngressClaim` to an `Ingress` is the responsibility of a
+process in the cluster.  A cluster may have any number of
+`LoadBalancerManager`s running, each of which handles some set of claim
+classes.  When an `IngressClaim` is created that asks for "Gold" ingress, the
+`LoadBalancerManager` which understands the "Gold" class will make the binding.
+
+`LoadBalancerManager`s can be simple `Pod`s running in the cluster, pulling
+cofiguration from hardcoded values, flags, config objects, or whatever
+mechanism makes sense for them.  We expect config objects to be a perfect fit
+for this use case.
```

Comment:
I would just say that it should be similar to the volume manager case rather than duplicate the text verbatim.